### PR TITLE
SEO: individualized content previews for DVC pages

### LINKED ISSUE #857: SEO: individualized content previews for DVC pages (and maybe /doc pages)
Currently preview of contents for all DVC pages has the same info:

image

Maybe it could have the actual title of each page at least? Ideally also a short excerpt of the beginning of the page contents.

### PR DESCRIPTION
**Pull request by MerlynMShelley**
Fixes #857

### CODE REVIEW

**Comment by rogermparent** [comment]
Hi @MerlynMShelley!

I see this is your first contribution: first, thanks for the fix! It looks good to me.  
I'd like to reformat this issue so it's more descriptive of what it contains and links the issue, however. No indictment on you, just that it'll make others have an easier time reading it.

Also, TypeScript lint is failing:

```
src/components/SEO/index.tsx:28:3 - error TS2339: Property 'site_name' does not exist on type 'PropsWithChildren<ISEOProps>'.

28   site_name,
```

Looks like you just have to add `site_name` to `ISEOProps` and that'll be solved.

**Comment by MerlynMShelley** [comment]
Yes, it is my first contribution. Okay, I get it about reformating. Thank you for your support and guidance.

File: src/components/SEO/index.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ const SEO: React.FC<ISEOProps> = ({
   title,
   defaultMetaTitle,
   description,
+  site_name,
```
Comment (shcheklein) [inline src/components/SEO/index.tsx:28]: site_name -> siteName (please, check the style around)

Comment (shcheklein) [reply src/components/SEO/index.tsx:28]: where does site name come from?

**Comment by shcheklein** [comment]
@MerlynMShelley thanks for the contribution! I don't see this fixing #857 though. We only added some additional meat attributes, right? I would say it partially addresses #1119 (not completely fixing it either).

Please, check some comments regarding var names, and it's not clear to me where do we get the value from, do we need some checks in place, etc

**Comment by MerlynMShelley** [comment]
@shcheklein Thank you for the support and guidance, I will add the camelCase structure and also include the custom text+ logo as you mentioned to have a distinct preview on the slack channel. I was just curious to resolve the issue. I would refer the links you mentioned, by Matt to come up with the best solution. Thank you.

File: src/components/SEO/index.tsx [hunk 2]
```diff
@@ -67,6 +68,22 @@ const SEO: React.FC<ISEOProps> = ({
       }
     )
   }
+  if (site_name) {
```
Comment (jorgeorpinel) [inline src/components/SEO/index.tsx:71]:
```suggestion

  if (site_name) {
```

**Comment by jorgeorpinel** [comment]
Thanks for this @MerlynMShelley! Roger and Ivan are already reviewing so please follow their lead in this PR. I'll just watch. Some notes from me:

Per https://github.com/iterative/dvc.org/pull/1379#issuecomment-636387293 please don't use the special word "Fixes" in the description of this PR, so merging this doesn't close that issue. Maybe say "Partially addresses #1119". alternatively, if you go for completely addressing that other issue, you can use "Fix #1119".

I also deployed this to https://dvc-landing-fix-hash-85-w68hes.herokuapp.com/ to check stuff and indeed there are no special previews:

image

But please use it to check the metatags, etc.

**Comment by jorgeorpinel** [comment]
Wait actually from my screenshot, the preview is indeed customized with the document's title, so this PR does seem to address #857 (as well as #1119 partially). Cc @shcheklein 

So if it's possible to also customize the snippet and/or image (per https://github.com/iterative/dvc.org/issues/857#issuecomment-564862662) then the description of this PR is accurate, I think.

**Comment by jorgeorpinel** [comment]
> the preview is indeed customized with the document's title

BUT it's missing the 2nd half of the page title, "| Data Version Control . DVC", @MerlynMShelley:

image

**Comment by MerlynMShelley** [comment]
Hi Team, @jorge, yes I will work on #1119 and #857. Thank you for your support and guidance. I will follow up the above said points.

File: src/components/SEO/index.tsx (hunk 1 above)
Comment (MerlynMShelley) [reply src/components/SEO/index.tsx]: siteName is one of the meta property as like siteTitle, so i have declared it on ISEO props and thereafter defined it below.

**Comment by shcheklein** [comment]
@MerlynMShelley a few questions:

1. I don't see `twitter:siteName` card here https://developer.twitter.com/en/docs/tweets/optimize-with-cards/guides/getting-started, neither I can't find the `og:siteName`. Could point us where did you find them?
2. Have you figured out what it the difference SEO vs DefaultSEO - it looks like site name is already defined in one of those. Could check one of the dvc.org pages? it might happen that we already site names there.
3. CircleCI is failing, it's something related to TS. Please, check it out and fix.

**Comment by MerlynMShelley** [comment]
I have referred this article on slack help,
https://slack.com/intl/en-in/help/articles/204399343-Share-links-and-set-preview-preferences

And I have performed site audit in these tools to find out what is missing in the site.

https://web.dev/measure/?gclid=CjwKCAjwztL2BRATEiwAvnALckrnPi64YihVniBJfNmlMZn-xnAHQ_VeOmYiHaGeeB4vXQsiHlqqcBoClh4QAvD_BwE

https://www.seoptimer.com/dvc.org#recommendation

https://seositecheckup.com/seo-audit/dvc.org

https://suite.seotesteronline.com/seo-checker/aHR0cDovL0R2Yy5vcmc=/content

Then I noticed that ```siteName``` was missing, also image ```alt``` text, schema.org ```structured test``` we're missing for rich media snippets. 

So I started by including the meta tags.

Yes, I will check according to your guidance and fix the same. Thank you.

**Comment by jorgeorpinel** [comment]
@MerlynMShelley wrong Jorge 🙂 

image

**Comment by MerlynMShelley** [comment]
Oh I am sorry.🙂. Will note that in future.

**Comment by jorgeorpinel** [comment]
@MerlynMShelley any updates on this? Thanks

**Comment by MerlynMShelley** [comment]
Yes, @jorgeorpinel.
https://github.com/iterative/dvc.org/pull/1379#issuecomment-637235511
Four days ago I mentioned this, so after adding siteName, we should add Schema to get rich media snippets. But in coding, I don't know to which file I have to make the addition. I can only suggest the coding part. But as of documentation, I have performed complete SEO site audit through site analytics tools like Neil Patel, SEMRush; it says the text content is very less compared to that of the coding part. So my project proposal would be to add more elaboratory SEO contents with keywords and LSI keywords to attain higher rankings in SERP. Thanks for asking.

**Comment by jorgeorpinel** [comment]
@MerlynMShelley I meant I don't think you've addressed all of Ivan's feedback in https://github.com/iterative/dvc.org/pull/1379#issuecomment-637162714 which would be necessary to move this PR forward.

As for the SEO report and your idea, sounds interesting but kind o unrelated to this PR. Please open a separate issue with the report and your proposal idea again so we can follow up there.

> we should add Schema to get rich media snippets. But in coding, I don't know to which file I have to make the addition. I can only suggest the coding part.

@rogermparent do you have any suggestions as for how this PR can be brought to a point were its mergeable, extracting the part Merlyn feels unable to contribute? Or perhaps provide specific guidance on what source code files to try editing? Thanks

**Comment by MerlynMShelley** [comment]
Ok, sure I follow what you have suggested @jorgeorpinel Thanks

**Comment by MerlynMShelley** [comment]
I don't find these ```Twitter card : creator``` and also the``` label/part1 ``` in the source code. As per the API doc, ```twitter:site:id twitter:player``` yet to be added in the source code I hope. But I don't know the exact place to add these in the source code.  Please refer here,  https://developer.twitter.com/en/docs/tweets/optimize-with-cards/overview/markup

Card Validator,
https://cards-dev.twitter.com/validator
@jorgeorpinel @shcheklein
Thanks

**Comment by MerlynMShelley** [comment]
Please find the SEO report of DVC.org,
https://www.seoptimer.com/www.dvc.org


Thanks, @jorgeorpinel @ shcheklein

**Comment by MerlynMShelley** [comment]
@shcheklein I found ```og:siteName``` meta property here,
https://megatags.co/#generate-tags

Secondly, I am not pretty much sure about where the changes have to be made in the src code
Thirdly CircleCI and TS are not familiar, yet to know about them.

As I propose to work on the SEO project idea In GSoD '20, I find this issue to make my first contributions.

Thank you.

**Comment by MerlynMShelley** [comment]
@shcheklein please also check this for ```siteName``` tag
https://ogp.me/

**Comment by rogermparent** [comment]
Sorry guys, I've been busy on another project and haven't checked my GitHub notifications in a bit.

> @rogermparent do you have any suggestions as for how this PR can be brought to a point were its mergeable, extracting the part Merlyn feels unable to contribute? Or perhaps provide specific guidance on what source code files to try editing? Thanks

By making the PR mergeable, do you mean just solving the CircleCI issue blocking merge or bringing the PR up to a standard of quality meant for merging?

for the former, here's the CI log:
```
$ tsc --noEmit --skipLibCheck && eslint --ext .json,.js,.ts,.tsx .
src/components/Page/DefaultSEO/index.tsx:21:29 - error TS2339: Property 'siteName' does not exist on type 'ISiteMeta'.
```
To fix this, `siteName` must be added to the `ISiteMeta` interface in `src/components/Page/DefaultSEO/index.tsx`

**Comment by shcheklein** [comment]
@MerlynMShelley 

og:siteName does not exist on that page, did you mean og:site_name - you should fix the code them

twitter:siteName does not exist.

@MerlynMShelley I'm really sorry, we don't have capacity to iterate with that much involvement on this particular task :( I'm going to close this and if you are still interested in this ticket please try to make a new PR taking into account all the feedback I left, and suggestion @rogermparent put together.

**Comment by jorgeorpinel** [comment]
@MerlynMShelley thanks for this and sorry it became complicated. It had a great start.

> Secondly, I am not pretty much sure about where the changes have to be made in the src code
Thirdly CircleCI and TS are not familiar, yet to know about them.
>
> As I propose to work on the SEO project idea In GSoD '20, I find this issue to make my first contributions.

The general feedback I can give you is that the contribution should be complete, not just starting changes that require the team to finish them. So if you don't feel comfortable with the code base, please focus on content for now.

Feel free to contribute with research about SEO issues and add some info in the issues, but please try to make it digested, not raw screenshots of some schema reference without links — those are pretty difficult to figure out. Please try to be brief and specific. However, actual contributions (PRs) would be best so again, maybe check our issues labeled `doc-content` + `good-first-issue` for now. Thanks again!

**Pull request closed without merging**